Apache Druid integration docs (#16426)
newrelic/docs-website: src/content/docs/infrastructure/host-integrations/host-integrations-list/apache-druid-integration.mdx

---
title: Apache Druid integration
tags:
  - Druid integration
  - apache druid integration
  - New Relic integrations
metaDescription: Use New Relic infrastructure agent to get a dashboard with druid metrics.
freshnessValidatedDate: 2024-04-21
---

import infrastructureDruidDashboard from 'images/infrastructure_screenshot-full_apache-druid-dashboard.webp'

> Integrating Apache Druid with New Relic enhances your ability to monitor, analyze, and optimize the performance of your Druid clusters. By leveraging New Relic's powerful monitoring and alerting features, you can ensure the reliability and efficiency of your Druid-based analytics platform.

Reviewer: "enhances" is a verb that i need to weave in sometimes...

Reviewer: https://docs.newrelic.com/docs/style-guide/word-choice/usage-dictionary/#New-Relic-brand-name 

We aren't supposed to make New Relic possessive unfortunately... maybe to simplify the language, I'd just say "This integration gives you a monitoring tool with alerting features, ensuring the reliability and blahblahblah"

> <img

Reviewer: I know fig captions are under debate, but i think a simple pathing caption might be good here. if only to add a little negative space padding around the path

    title="Apache Druid dashboard"
    alt="Apache Druid dashboard"
    src={infrastructureDruidDashboard}
/>

## Set up the Apache Druid integration

Complete the following steps to set up the Apache Druid integration:

<Steps>
    <Step>
## Install the infrastructure agent

> To use the Apache Druid integration, you need to also [install the infrastructure agent](/docs/infrastructure/install-infrastructure-agent/get-started/install-infrastructure-agent-new-relic/) on the same host. The infrastructure agent monitors the host itself, while the integration you'll install in the next step extends your monitoring with Apache Druid-specific data.    

Reviewer: ```suggestion
To use the Apache Druid integration, you need to first [install the infrastructure agent](/docs/infrastructure/install-infrastructure-agent/get-started/install-infrastructure-agent-new-relic/) on the same host. The infrastructure agent monitors the host itself, while the Apache Druid integration extends your monitoring with data specific to your Druid clusters.    
```

Reviewer: this nested clause is a little bit of a mouthful: integration you'll install in the next step 

am offering some alt wording for readability, but it's just an option. :)

    </Step>
    <Step>

> ## Expose Druid metrics using prometheus emitter [#Expose]

Reviewer: When referring to the concept of `prometheus.emitter`, does Prometheus need capitalization? Should we stylize this with backticks?

1. Add `prometheus.emitter` to the end of the extensions load list in your `apache-druid-24.0.0/conf/druid/single-server/micro-quickstart/_common/common.runtime.properties` file:

```yml
druid.extensions.loadList=["druid-hdfs-storage", "druid-kafka-indexing-service", "druid-datasketches", "druid-multi-stage-query", "prometheus-emitter"]
```

> 2. Add the code snippets in the right column to the files in the left column:

Reviewer: I would make this sentence parallel left --> right of the table, so either add the snippets to the left column and file paths to the right (Add the code snippets in the left column to the file paths in the right column), or change the wording to: 

"In the file paths listed in the left column, add the code snippets listed in the right columns."

Reviewer: i know this is a huge pain but the parallelism will def help

druid.emitter.prometheus.port=19093
```
            </td>
        </tr>
        <tr>
            <td>
`PATH/TO/MiddleManager.runtime.properties`
            </td>
            <td>
```yml
# Monitoring
druid.monitoring.monitors=["org.apache.druid.java.util.metrics.JvmMonitor"]
druid.emitter=prometheus
druid.emitter.logging.logLevel=info
druid.emitter.prometheus.strategy=exporter
druid.emitter.prometheus.port=19094
```
            </td>
        </tr>
        <tr>
            <td>
`PATH/TO/Router.runtime.properties`
            </td>
            <td>
```yml
# Monitoring
druid.monitoring.monitors=["org.apache.druid.java.util.metrics.JvmMonitor"]
druid.emitter=prometheus
druid.emitter.logging.logLevel=info
druid.emitter.prometheus.strategy=exporter
druid.emitter.prometheus.port=19095
```
            </td>
        </tr>
    </tbody>
</table>

    </Step>
    <Step>
## Install the Prometheus Emitter Extension [#prometheus-emitter-extension]

1. Run the following commands to create a folder named `prometheus-emitter` within the `extensions` folder directory of your Apache Druid setup:
```shell
cd apache-druid-29.0.0/extensions/
```
```shell
sudo mkdir prometheus-emitter
```

> 2. Navigate to the druid download directory and run the following command to create jar files that will be called when the server starts up:

Reviewer: ```suggestion
2. Navigate to the druid download directory and run the following command to create jar files that the server calls on start up:
```

```shell
java \
  -cp "lib/*" \
  -Ddruid.extensions.directory="extensions" \
  -Ddruid.extensions.hadoopDependenciesDir="hadoop-dependencies" \
  org.apache.druid.cli.Main tools pull-deps \
  --no-default-hadoop \
  -c "org.apache.druid.extensions.contrib:prometheus-emitter:24.0.0"
```   
    </Step>
    <Step>

> ## Configure `nri-prometheus` for Apache Druid [#configure]

Reviewer: for steps related to new relic specific files/configurations, i might add a brief, standard/repeatable explanation about how this yml file organizes the directory or something to that effect. it doesn't have to be verbose or anything, maybe something like 'Create a file named `nri-prometheus-config.yml`. This file directs X to collect Y metrics, so it becomes part of our ingest pipeline.' idk how it works i'm just riffing

Reviewer 2: Great idea, but this will have to be a fast follow. I'll consult with SMEs!
